src/components/editor/metaform-editor-right-drawer.tsx

```diff
@@ -211,33 +236,36 @@ const MetaformEditorRightDrawer: React.FC<Props> = ({
     }
     if (fieldIndex !== undefined && sectionIndex !== undefined) {
       const field = pendingForm.sections![sectionIndex].fields![fieldIndex];
-      if (requiredField === undefined) {
+      if (requiredField === "") {
         return null;
       }
-      if (requiredField !== undefined) {
+      if (requiredField !== "") {
         return (
-          <FormControl fullWidth>
-            <InputLabel id="visiblityConditionLabel">{ strings.draftEditorScreen.editor.visibility.conditionalFieldValue }</InputLabel>
-            <Select
-              fullWidth
-              labelId="visiblityConditionLabel"
-              label={ strings.draftEditorScreen.editor.visibility.conditionalFieldValue }
-              value=""
-              onChange={ event => updateFormField({
-                ...field,
-                visibleIf: {
-                  field: requiredField,
-                  equals: event.target.value,
-                  notEquals: "",
-                  and: [],
-                  or: []
-                }
-              }) }
-            >
-              <MenuItem value="kylla">{ strings.generic.yes }</MenuItem>
-              <MenuItem value="ei">{ strings.generic.no }</MenuItem>
-            </Select>
-          </FormControl>
+          <>
+            <FormControl fullWidth>
+              <InputLabel id="visiblityConditionLabel21">{ strings.draftEditorScreen.editor.visibility.conditionalFieldValue }</InputLabel>
+              <Select
+                fullWidth
+                labelId="visiblityConditionLabel21"
+                label={ strings.draftEditorScreen.editor.visibility.conditionalFieldValue }
+                value={ requiredFieldBoolean }
+                onChange={ event => updateFormFieldVisiblityValues({
+                  ...field,
+                  visibleIf: {
+                    field: requiredField,
+                    equals: event.target.value,
+                    notEquals: "",
+                    and: [],
+                    or: []
+                  }
+                }, event.target.value) }
+              >
+                <MenuItem value="kylla">{ strings.generic.yes }</MenuItem>
+                <MenuItem value="ei">{ strings.generic.no }</MenuItem>
+              </Select>
+            </FormControl>
+            <Divider/>
+          </>
```

Comment:
Divider -component could be moved to components return and therefore this methods return doesn't need to be wrapped in Fragments.